packages/console/src/pages/Main/index.tsx

```diff
@@ -68,83 +42,15 @@ const Main = () => {
-                  />
-                  <Route path=":tab" element={<SignInExperience />} />
-                </Route>
-                <Route path="connectors">
-                  <Route index element={<Navigate replace to={ConnectorsTabs.Passwordless} />} />
-                  <Route path=":tab" element={<Connectors />} />
-                  <Route path=":tab/create/:createType" element={<Connectors />} />
-                  <Route path=":tab/:connectorId" element={<ConnectorDetails />} />
-                </Route>
-                <Route path="users">
-                  <Route index element={<Users />} />
-                  <Route path="create" element={<Users />} />
-                  <Route path=":id" element={<UserDetails />}>
-                    <Route index element={<Navigate replace to={UserDetailsTabs.Settings} />} />
-                    <Route path={UserDetailsTabs.Settings} element={<UserSettings />} />
-                    <Route path={UserDetailsTabs.Roles} element={<UserRoles />} />
-                    <Route path={UserDetailsTabs.Logs} element={<UserLogs />} />
-                  </Route>
-                  <Route
-                    path={`:id/${UserDetailsTabs.Logs}/:logId`}
-                    element={<AuditLogDetails />}
-                  />
-                </Route>
-                <Route path="audit-logs">
-                  <Route index element={<AuditLogs />} />
-                  <Route path=":logId" element={<AuditLogDetails />} />
-                </Route>
-                <Route path="roles">
-                  <Route index element={<Roles />} />
-                  <Route path="create" element={<Roles />} />
-                  <Route path=":id" element={<RoleDetails />}>
-                    <Route index element={<Navigate replace to={RoleDetailsTabs.Settings} />} />
-                    <Route path={RoleDetailsTabs.Settings} element={<RoleSettings />} />
-                    <Route path={RoleDetailsTabs.Permissions} element={<RolePermissions />} />
-                    <Route path={RoleDetailsTabs.Users} element={<RoleUsers />} />
-                  </Route>
-                </Route>
-                <Route path="profile">
-                  <Route index element={<Profile />} />
-                  <Route path="verify-password" element={<VerifyPasswordModal />} />
-                  <Route path="change-password" element={<ChangePasswordModal />} />
-                  <Route path="link-email" element={<LinkEmailModal />} />
-                  <Route path="verification-code" element={<VerificationCodeModal />} />
-                </Route>
-              </Route>
+            <Route element={<AppContent />}>
+              {isOnboarding ? (
+                <Route>
+                  <Route index element={<Navigate replace to={`/${CloudRoute.Onboard}`} />} />
+                  <Route path={`/${CloudRoute.Onboard}/*`} element={<Onboard />} />
+                </Route>
```

Comment:
i'm thinking the onboarding is actually a cloud-only flow and should not be coupled with all these oss paths.

it should be deviated from an upper level. should be something like:

```tsx
const Component = () => {
  // ONLY necessary hooks and variables

  if (some condition check) {
    return <Onboarding />
  }

  return <Main />;
};
```